docs/developer/architecture/snmp.md

```diff
@@ -1,3 +1,102 @@
 # SNMP
 
------
+!!! note
+    This section is meant for developers that want to understand the working of the SNMP integration.
+
+    Be sure you are familiar with [SNMP concepts](../../tutorials/snmp/introduction/), and you have read through the official [SNMP integration docs](https://docs.datadoghq.com/integrations/snmp).
+
+## Overview
+
+While most integrations are either Python, JMX, or implemented in the Agent in Go, the SNMP integration is a bit more complex.
+
+Here's an overview of what this integration entails:
+
+* A [Python check](https://github.com/DataDog/datadog-agent/blob/master/snmp), responsible for:
+    * Collecting metrics from a specific device IP.
+    * Auto-discovering devices over a network. (Pending deprecation in favor of Agent auto-discovery.)
+* An [Agent service listener](https://github.com/DataDog/datadog-agent/blob/master/pkg/autodiscovery/listeners/snmp.go), responsible for auto-discovering devices over a network and forwarding discovered instances to the existing Agent check scheduling pipeline. Also known as "Agent SNMP auto-discovery".
+
+As a diagram:
+
+![](/assets/images/snmp-architecture.png)
```

Comment:
> AFAIK main intended use case is with DCA.

It's not that clear to me. If the user doesn't have any infra on Kubernetes and/or a single Agent is enough to collect metrics from a few devices, then Agent auto-discovery on a single Agent can still be useful.

Reply:
> It's not that clear to me. If the user doesn't have any infra on Kubernetes and/or a single Agent is enough to collect metrics from a few devices, then Agent auto-discovery on a single Agent can still be useful.

Yes, true, we still get some benefits using Agent AD with a single server over Python SNMP AD:
We leverage Agent Scheduling of python checks (one check instance per device, instead of one check instance only for Python snmp AD).

Reply:
> We leverage Agent Scheduling of python checks (one check instance per device, instead of one check instance only for Python snmp AD).

This looks like the standard behaviour for Python-based integrations, which is why I didn't mention it explicitly and instead focused on showing how Python SNMP AD is the odd one out ("Caveats" section).